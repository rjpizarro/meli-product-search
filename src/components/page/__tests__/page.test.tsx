import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import Page from '../index'

describe('Page', () => {
    it('should render correctly ', () => {
        const tree = renderer
            .create(<Page>
                <p>child</p>
            </Page>)
            .toJSON()
        const { getByText } = render(
            <Page>
                <p>child</p>
            </Page>
        )

        expect(tree).toMatchSnapshot()
        expect(getByText('child')).toBeInTheDocument()
    })

    it('should extend classNames when receive a className prop', () => {
        const { container } = render(
            <Page className="find-this-class">
                <p>child</p>
            </Page>
        )

        expect(container.querySelector('[class="find-this-class"]')).toBeTruthy()
    })
})

