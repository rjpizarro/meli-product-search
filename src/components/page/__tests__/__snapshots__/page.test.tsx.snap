// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page should render correctly  1`] = `
<main>
  <div
    className="page container"
  >
    <div
      className="row center-xs"
    >
      <div
        className="col-xs-10 page__content"
      >
        <p>
          child
        </p>
      </div>
    </div>
  </div>
</main>
`;
